src/get-requested-version.ts

```diff
@@ -0,0 +1,52 @@
+import * as core from "@actions/core";
+
+import { readFile } from "fs";
+import { join } from "path";
+
+const semverRegex = /^((?:\d+)(?:\.\d+){0,2})$/;
+
+async function readVersionFromFile(filePath: string) {
+  let version = "";
+
+  readFile(filePath, "utf8", (err, data) => {
+    if (err != null) throw err;
+
+    const parsedVersion = data.match(semverRegex)?.at(1);
```

Comment:
There is a semver package that's already used in the project that you can use to validate the string 🙂